server_src/runners/history.js

```diff
@@ -25,15 +23,284 @@ const script = require("./scriptCheck.js");
 const { ScriptRunner } = script;
 
 const MjpegDecoder = require("mjpeg-decoder");
-const { downloadFromOctoPrint } = require("../utils/download.util");
 
 const { writePoints } = require("../lib/influxExport.js");
 
-const logger = new Logger("OctoFarm-HistoryCollection");
 let counter = 0;
 let errorCounter = 0;
 
+const {
+  generateHistoryIndex,
+  generateHistoryState,
+  generateHistoryReason,
+  generateHistoryFileInformation,
+  generateHistoryDate,
+  generateHistoryPrintTime,
+  generateHistoryResendStatistics,
+  generateHistoryJobStatistics,
+} = require("../services/history.service.js");
+
+const { definePrinterName } = require("../services/printer.service.js");
+
+const {
+  checkForValidPrinterCostSettings,
+} = require("../services/printer-settings.service.js");
+
+const {
+  octoprintGrabThumbnail,
+} = require("../services/octoprint/octoprint-plugins.service.js");
+
+const {
+  objectIDValidator,
+  generateObjectId,
+} = require("../utils/database.utils.js");
+
+class History {
+  _id;
```

Comment:
Love this explicit property setting. Think it will fit V2 nicely if it works.

Reply:
Oh! Was unsure about those!